src/cobol/koopa/cobol/parser/preprocessing/replacing/ReplacingPhraseOperand.java

```diff
@@ -104,6 +104,15 @@ private void prepareForMatching() {
 			atSpace = false;
 		}
 	}
+	
+	private boolean isDummyOperand() {
```

Comment:
This assumes that a dummy operand is the only thing to appear in a pseudo literal. E.g. you can't have `==:TAG:-A==` or `==B-:TAG:==`. Is this a valid assumption ? The [link you shared](https://www.ibm.com/support/knowledgecenter/en/SS6SG3_4.2.0/com.ibm.entcobol.doc_4.2/PGandLR/ref/rlcdscop.htm) does not specify this. Is there a more complete reference on the rules for dummy operands ?

Reply:
I can't find IBM's language specification, only the language reference I sent. I have run some adhoc tests using IBM Enterprise COBOL for z/OS  4.2.0 and it seems to just get more confusing.

The cases of `==:TAG:-A==` or `==B-:TAG:==` does not work in the same way. It looks like those cases are treated in the same way as REPLACING without the dummy operand. I observed that the pattern was replaced at the beginning of a word but not in the middle of the word. It looks like the dummy operand was introduced to allow for replacing in the middle of words.

I found the following in an old message board 

> It is a common misconception that COPY REPLACING can replace parts of COBOL variables. This is a misconception because COBOL only allows replacement of entire words -- unless a delimiting character such as a colon is used. And COBOL considers the delimiting characters to create a word.

It looks like the pattern of `==:TAG:==` is a special case that should be handled differently.

I'll update ReplacingPhrase.java with the name changes and also enforce the requirement that the dummy operand must start and end with colons, like I did in ReplacingPhraseOperand.java. I'll also add your tests and figure out why they are failing.

Reply:
> I'll also add your tests and figure out why they are failing.

Hm, all tests are passing here. How do you trigger the tests ?

Reply:
They are working here as well. I misread your comment about the test failing against koopa master and read it as tests failing in this pull request.

Reply:
The change also has an assumption that there is a closing colon (exactly 2 colons) in ReplacingPhase.nonBlankProgramText(). This method will return when it gets a blank token, so the risk of reading far ahead is small, but it means that replace won't work if your replace contains a colon.

Reply:
Could you give an example of such a replace, and what it should (or shouldn't do) ? If we can handle it in this fix, that would be great. If it's something worthy of its own PR later, then that's fine too. (But in that case we could make an issue for it.)

Reply:
Since I don't have the specification, I can't be sure, but I have the following observations.

For IBM's preprocessor `==PFX:==` will do the replace. This does not work in koopa, but I get the feeling that this isn't the intention from IBM and could be a bug.

For IBM's preprocessor `==:PFX==` does not do the replacement. Koopa doesn't do the replacement either.

For IBM's preprocessor `==P:FX==` does not do the replacement. Koopa doesn't do the replacement either.

I don't know if you are going after specific flavours of COBOL, but it doesn't seem like IBM is consistent on this one. These are just some simple cases and there could be even more, for example, `==PFX:::==`. I don't think it's worth trying to special case this right now.

Reply:
Oh, very interesting. Thanks for clarifying.

I try to keep the COBOL parser general, at least in the sense that I would prefer not to make flavour-specific variants of the grammar. I want to be able to throw any (valid) COBOL code at it and get a parse. That does mean I have to support flavour-specific constructs from time to time, and if you look at the COBOL grammar you'll see evidence of that.

I agree that there is no good reason right now to support the oddity in IBM's preprocessor. We can revisit this if there turns out to be a big valid set of COBOL code which does rely on it.